.github/workflows/test.yml

```diff
@@ -64,13 +69,22 @@ jobs:
         with:
           python-version: '3.8'
 
-      - name: Setup Python package dependencies
+      - name: Cache pip
+        uses: actions/cache@v2
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-python-${{ hashFiles('**/*requirements.txt') }}
+          restore-keys: |
+            ${{ runner.os }}-python-${{ hashFiles('**/*requirements.txt') }}
+            ${{ runner.os }}-python-
+
+      - name: Update pip
         run: |
-            # Install development requirements, package requirements, and frozen image
-            # requirements, with an increasing priority.
-            pip install -r dev-requirements.txt
-            pip install .
-            pip install -r helm-chart/images/binderhub/requirements.txt
+            pip install --upgrade pip
```

Comment:
Package manager as well as libraries used for installing dependencies should always be upgraded to the most recent version.